packages/core/src/modules/connections/ConnectionsApi.ts

```diff
@@ -227,6 +230,42 @@ export class ConnectionsApi {
     return connectionRecord
   }
 
+
+  /**
+   * Send a trust ping to an established connection
+   *
+   * @param connectionId the id of the connection for which to accept the response
+   * @param responseRequested do we want a response to our ping
+   * @param withReturnRouting do we want a response at the time of posting
+   * @returns boolean indicating success or failure
+   */
+  public async trustPing(connectionId: string, responseRequested=true, withReturnRouting=false): Promise<boolean> {
+    const connectionRecord = await this.connectionService.getById(this.agentContext, connectionId)
+
+    let outboundMessageContext
+    if (connectionRecord.state === DidExchangeState.Completed) {
+
+      const { message } = await this.connectionService.createTrustPing(this.agentContext, connectionRecord, {
+        responseRequested: responseRequested,
+      })
+      // Disable return routing as we don't want to receive a response for this message over the same channel
+      // This has led to long timeouts as not all clients actually close an http socket if there is no response message
+      if (!withReturnRouting) {
+        message.setReturnRouting(ReturnRouteTypes.none)
+      }
```

Comment:
Where are you using the trust ping for? I'm thinking whether we can come up with a general way of doing this. No other method in the API of AFJ has this option, and it feels a bit use case specific specific. Thinking how we can generically solve this without needing to add it to all other messages. Also because this seems more of an issue of the other agents, rather than an issue of AFJ. We just instruct the other agent that they _may_ return it over the same channel. For ACA-Py this has been fixed in the main branch, but in any release yet sadly: https://github.com/hyperledger/aries-cloudagent-python/pull/1853

Could we base it off the `responseRequested` parameter maybe for now?

Reply:
The trust ping can be used as a diagnostic tool. Being able to specify the return route allows us to perform different diagnostics. For example, we can test the return route is behaving as expected.

In the particular use case, I'm trying to ensure that the websocket of the mediator is responding correctly, and test the latency of the mediator.

Reply:
@reflectivedevelopment -- are you working on figuring out a scalable mediator (Aries in general) deployment?  E.g. persistent queue/shared cache enabled with support for websockets and return-route?  We're working on that, and would like to collaborate on findings/recommendations.

Reply:
If that's the case you may be able to get away with a custom plugin / module that hooks into the metal of AFJ?

Diagnostics / Latency tests seem like an advanced use case to me. 

What I'm getting at is that I want to be careful with adding every needed property to a specific method when a single person needs it. This makes the API inconsistent, and adds complexity where it may not be needed. That said, if you think it is generally useful to have this, I'm okay with adding such an option -- but as I said, it feels like an advanced use case that may deserve it's own method / api

Reply:
I think supporting testing frameworks is an important aspect of the AFJ framework, and seeing that there is little overhead in this change, it seems the most reasonable approach to me.